interfaces/hotplug/deviceinfo.go

```diff
@@ -89,3 +89,38 @@ func (h *HotplugDeviceInfo) Attribute(name string) (string, bool) {
 	val, ok := h.Data[name]
 	return val, ok
 }
+
+func (h *HotplugDeviceInfo) firstAttrValueOf(tryAttrs ...string) string {
+	for _, attr := range tryAttrs {
+		if val, _ := h.Attribute(attr); val != "" {
+			return val
+		}
+	}
+	return ""
+}
+
+func (h *HotplugDeviceInfo) String() string {
+	var nameOrPath string
+
+	// devname is the name of the device under /dev, eg. /dev/ttyS0;
+	// prefer devname over devpath as this is the one used to talk to the device.
+	if nameOrPath = h.DeviceName(); nameOrPath == "" {
+		// devpath is the path of the device under /sys, eg. /sys/devices/pnp0/00:04/tty/ttyS0.
+		nameOrPath = h.DevicePath()
+	}
+
+	modelOrVendor := h.firstAttrValueOf("ID_MODEL_FROM_DATABASE", "ID_MODEL", "ID_MODEL_ID", "ID_VENDOR_FROM_DATABASE", "ID_VENDOR", "ID_VENDOR_ID")
+	if modelOrVendor == "" {
+		modelOrVendor = "?"
+	}
+	if len(modelOrVendor) > 16 {
+		modelOrVendor = modelOrVendor[0:16] + "..."
```

Comment:
it's a bit strange to cut this at 16  but let through the 65 chars hotplug key. We should start by not truncating for now, or truncating at 70

Reply:
is our worry about length the task summaries?

maybe we need a specific method for those that does the more aggressive truncating, also in those we should probably cut the hotplug key  to  [:12]+"..."

Reply:
Yes, the concern is task/change summaries. I've upped String() to 70 and added ShortString() that truncates at 12, for use in task/summaries.

As for hotplug key I'd like to do this in a followup - I think that possibly the best solution would be to have a real type of hotplug key (just string type alias maybe), and proper String() for it; this will also improve readability of all the methods that currently take string arg for the hotplug key.